x-pack/filebeat/input/azureblobstorage/job.go

```diff
@@ -208,6 +215,10 @@ func (j *job) decode(ctx context.Context, r io.Reader, id string) error {
 		}
 	}
 
+	if err != nil {
+		j.status.UpdateStatus(status.Degraded, err.Error())
+	}
```

Comment:
The only way that we can reach this is from the decode loop. I think probably this should be put there in the place of the break and have a return nil at the bottom of the function to make error paths clearer here.